### PR TITLE
Prometheus: make JS heatmap response a single-frame (field-per-bucket)

### PR DESCRIPTION
**Pull request by leeoniya**
...instead of multi-frame (frame-per-bucket)

part of https://github.com/grafana/grafana-prometheus-datasource/issues/151

i don't have a local prometheus instance to verify if any of this actually continues to produce correct heatmaps, please help me test :)

### CODE REVIEW

**Comment by aocenas** [comment]
@leeoniya you can use the devenv data sources: https://github.com/grafana/grafana/tree/main/devenv should be fairly easy to setup.

Also I don't think we removed the direct access in prometheus yet, so not sure we can remove the front end code at this moment, that should also be checked.

**Comment by ryantxu** [comment]
Before:
image

With query:
image

After:
image

Looks good to me -- but good if someone from @grafana/observability-metrics can take a look

File: public/app/plugins/datasource/prometheus/result_transformer.ts [hunk 1]
```diff
@@ -533,6 +535,33 @@ export function getOriginalMetricName(labelData: { [key: string]: string }) {
...
+    return [];
+  }
+
+  const timeField = frames[0].fields.find((field) => field.type === FieldType.time)!;
+  const countFields = frames.map((frame) => {
+    let field = frame.fields.find((field) => field.type === FieldType.number)!;
+
+    return {
+      ...field,
+      name: field.config.displayNameFromDS!,
```
Comment (toddtreece) [inline public/app/plugins/datasource/prometheus/result_transformer.ts:549]: should this be the frame name?

Comment (leeoniya) [reply public/app/plugins/datasource/prometheus/result_transformer.ts:549]: i think they're always the same? `displayNameFromDS` felt less arbitrary / more explicit.

**Review by toddtreece — APPROVED** [review]
it seems like there is still an issue with the prom heatmap format in the new heatmap panel. after talking with @leeoniya, it sounds like the approach will be to merge this since it doesn't break the old heatmap panel, and fix the new panel in a followup PR

**Pull request merged**